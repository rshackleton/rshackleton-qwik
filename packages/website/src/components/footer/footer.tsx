import { component$ } from '@builder.io/qwik';

export type FooterProps = {};

export const Footer = component$<FooterProps>(() => {
  return (
    <footer class="flex place-content-center place-items-center items-center px-4 py-8">
      <p class="text-sm text-neutral-500">
        &copy; Richard Shackleton {new Date().getUTCFullYear()}
      </p>
    </footer>
  );
});
