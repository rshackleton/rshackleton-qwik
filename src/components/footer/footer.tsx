import { component$ } from '@builder.io/qwik';

export const Footer = component$(() => {
  return (
    <footer class="">
      <div class="site-container py-4">
        <p class="text-sm">&copy; Richard Shackleton</p>
      </div>
    </footer>
  );
});
