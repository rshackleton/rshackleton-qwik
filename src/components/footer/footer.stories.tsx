import type { Meta, StoryObj } from 'storybook-framework-qwik';
import type { FooterProps } from './footer';
import { Footer } from './footer';

export default {
  title: 'Footer',
  component: Footer,
} as Meta<FooterProps>;

export const Default: StoryObj<FooterProps> = {};
