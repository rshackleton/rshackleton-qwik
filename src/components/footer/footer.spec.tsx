import { createDOM } from '@builder.io/qwik/testing';
import { expect, test } from 'vitest';
import { Footer } from './footer';

test(`[Footer Component]: Renders without error`, async () => {
  const { screen, render } = await createDOM();
  await render(<Footer />);
  expect(screen.outerHTML).toMatchSnapshot();
});
