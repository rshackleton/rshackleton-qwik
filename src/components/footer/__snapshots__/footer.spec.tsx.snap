// Vitest Snapshot v1

exports[`[Footer Component]: Renders without error 1`] = `"<host q:version=\\"0.16.2\\" q:container=\\"resumed\\" q:render=\\"dom-dev\\"><!--qv q:key=hg9U:yE_0 data-qrl=Footer_component_hg9USy095T4--><footer class=\\"mx-auto max-w-screen-2xl px-8 py-8\\"><p class=\\"text-sm\\">Â© Richard Shackleton</p></footer><!--/qv--></host>"`;
